File: haskell/src/Common_Lib.hs
module Common_Lib (copyFileTo, copyHandleTo) where

import qualified Data.ByteString as BS
import System.IO (Handle)

copyHandleTo :: Handle -> Handle -> IO ()
copyHandleTo input output = do
  bs <- BS.hGetContents input
  BS.hPut output bs

copyFileTo :: FilePath -> Handle -> IO ()
copyFileTo path output = do
  bs <- BS.readFile path
  BS.hPut output bs

File: swift/src/common_lib.swift
import Foundation

enum common_lib {
    static func copyStream(_ input: FileHandle, _ output: FileHandle) throws {
        while true {
            let data = input.readData(ofLength: 65536)
            if data.isEmpty { return }
            output.write(data)
        }
    }

    static func copyFile(_ path: String, _ output: FileHandle) throws {
        let handle = try FileHandle(forReadingFrom: URL(fileURLWithPath: path))
        defer { try? handle.close() }
        try copyStream(handle, output)
    }
}
